opennms-provision/opennms-provisiond/src/test/java/org/opennms/netmgt/provision/service/ProvisionerIT.java

```diff
@@ -1797,6 +1803,52 @@ public void testRequisitionedCategoriesThenUpdateRequisitionToRemoveCategory() t
         assertEquals(0, n.getCategories().size());
     }
 
+    @Test(timeout = 300000)
+    public void testRequisitionedCategoriesThenRemoveCategoryByApi() throws Exception {
+        final int nextNodeId = m_nodeDao.getNextNodeId();
+
+        importFromResource("classpath:/provisioner-testCategories-ThreeCategories.xml", Boolean.TRUE.toString());
+
+        OnmsNode n = getNodeDao().get(nextNodeId);
+        assertEquals(3, n.getCategories().size());
+        assertTrue(n.hasCategory("NMS-16536"));
+
+        runPendingScans();
+
+        assertEquals(3, m_categoryAssociationDao.findAll().size());
+
+        OnmsCategory deleteCategory = n.getCategories().stream()
+                .filter(cat -> cat.getName().equals("NMS-16536")).findFirst().orElse(null);
+        m_categoryDao.delete(deleteCategory);
+        m_categoryDao.flush();
+
+        assertEquals(2, m_categoryDao.findAll().size());
+
+        RequisitionedCategoryAssociation deleteCategoryAssociation = m_categoryAssociationDao.findAll().stream()
+                .filter(cat -> cat.getCategory().getName().equals("NMS-16536")).findFirst().orElse(null);
+        m_categoryAssociationDao.delete(deleteCategoryAssociation);
+        m_categoryAssociationDao.flush();
+
+        assertEquals(2, m_categoryAssociationDao.findAll().size());
+
+        m_provisionService.updateNodeAttributes(n);
+
+        assertEquals(3, m_categoryAssociationDao.findAll().size());
```

Comment:
As I understand, `m_provisionService.updateNodeAttributes(n)` shouldn't add categories back, right ?   It should update cache based  by loading dao again. What is missing here ?

Reply:
As per the flow, when a new category is added via the UI, it is stored in the database if it does not already exist. In the current scenario, the category was deleted from the database but is still being sent from the UI. Therefore, this assert ensures that association is added to the requisition categories.